Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm

```diff
@@ -297,11 +312,15 @@ - (void)authorizationController:(ASAuthorizationController *)controller didCompl
     m_controller = WTFMove(controller);
     m_completionHandler = WTFMove(handler);
     m_delegate = adoptNS([[_WKASDelegate alloc] initWithPage:WTFMove(requestData.page) completionHandler:makeBlockPtr([this](ASAuthorization *auth, NSError *error) mutable {
-        ensureOnMainRunLoop([this, auth = retainPtr(auth)]() {
+        ensureOnMainRunLoop([this, auth = retainPtr(auth), error = retainPtr(error)]() {
```

Comment:
@pascoej Shouldn't we capture a weakThis here? How can we guarantee that `this` is still alive when the lambda runs asynchronously?

Reply:
Changed to weakThis